AngularJS/src/components/registration/registration.component.js

```diff
@@ -2,7 +2,24 @@ const angular = require('angular');
 
 angular.module('app-bootstrap').component('registration', {
   template: require('./registration.pug')(),
-  controller: [function () {
-    this.component2Phrase = 'This is registration 2';
+  controller: ['userService', '$state', function (userService, $state) {
+    this.submitted = false;
+    this.user = {
+      name: '',
+      lastname: '',
+      password: '',
+      email: ''
+    };
+    this.error = {
+      name: false,
+      lastname: false,
+      password: false,
+      email: false
+    };
+    this.createUser = () => {
+      this.submitted = true;
+      userService.create(this.user);
```

Comment:
The `userService` should be all returning Promises by using the `$http` to make requests to the API